src/utils/functionSignatureParser.peg

```diff
@@ -0,0 +1,240 @@
+      if (input instanceof Array) {
+        error(`Expected input to be number: ${input}`);
+        throw "error already raised, appeasing the typesystem";
+      }
+      let i: bigint;
+      try {
+        i = BigInt(input);
+      } catch {
+        error(`Expected input to be number: ${input}`);
+        throw "error already raised, appeasing the typesystem";
+      }
+      if (i >= 2n ** BigInt(width)) error(`Uint${width} exceeds bound: ${input}`);
+      if (i < 0n) error(`Uint${width} is less than zero: ${input}`);
+      return toUintOrFelt(i, width);
+    }
+  }
+  / "int" width:Integer { return (input: Input) => {
+      if (input instanceof Array) {
+        error(`Expected input to be number: ${input}`);
+        throw "error already raised, appeasing the typesystem";
+      }
+      let i: bigint;
+      try {
+        i = BigInt(input);
+      } catch {
+        error(`Expected input to be number: ${input}`);
+        throw "error already raised, appeasing the typesystem";
+      }
+      if (i >= 2n ** BigInt(width - 1)) error(`Int${width} exceeds bound: ${input}`);
+      if (i < -(2n ** BigInt(width - 1))) error(`Int${width} exceeds bound: ${input}`);
+      return toUintOrFelt(i, width);
+    }
+  }
+  / "uint" { return (input: Input) => {
+      if (input instanceof Array) {
+        error(`Expected input to be number: ${input}`);
+        throw "error already raised, appeasing the typesystem";
+      }
+      let i: bigint;
+      try {
+        i = BigInt(input);
+      } catch {
+        error(`Expected input to be number: ${input}`);
+        throw "error already raised, appeasing the typesystem";
+      }
+      if (i >= 2n ** BigInt(256n)) error(`Uint256 exceeds bound: ${input}`);
+      if (i < 0n) error(`Uint256 is less than zero: ${input}`);
+      return toUintOrFelt(i, 256);
+    }
+  }
+  / "int" { return (input: Input) => {
+      if (input instanceof Array) {
+        error(`Expected input to be number: ${input}`);
+        throw "error already raised, appeasing the typesystem";
+      }
+      let i: bigint;
+      try {
+        i = BigInt(input);
+      } catch {
+        error(`Expected input to be number: ${input}`);
+        throw "error already raised, appeasing the typesystem";
+      }
+      if (i >= 2n ** BigInt(255n)) error(`Int256 exceeds bound: ${input}`);
+      if (i < -(2n ** BigInt(255n))) error(`Int256 exceeds bound: ${input}`);
+      return toUintOrFelt(i, 256);
+    }
+  }
+  / "address" { return (input: Input) => {
+      if (input instanceof Array) {
+        error(`Expected input to be number: ${input}`);
+        throw "error already raised, appeasing the typesystem";
+      }
+      let i: bigint;
+      try {
+        i = BigInt(input);
+      } catch {
+        error(`Expected input to be number: ${input}`);
+        throw "error already raised, appeasing the typesystem";
+      }
+      if (i >= 2n ** BigInt(251)) error(`Address exceeds bound: ${input}`);
```

Comment:
The starknet field prime is slightly bigger than 2 ** 251.
FIELD_PRIME = 2 ** 251 + 17 * 2 ** 192 + 1
https://github.com/starkware-libs/cairo-lang/blob/cf8266fd5d1ff66962579ff7967ac5cdcf699f77/src/starkware/crypto/starkware/crypto/signature/nothing_up_my_sleeve_gen.py#L35

Reply:
Turns out addresses don't take the the full felt space